hummingbird/ml/operator_converters/_base_operator.py

```diff
@@ -0,0 +1,15 @@
+from abc import ABC
+
+
+class BaseOperator(ABC):
+    """
+    Abstract class definig the basic structure for operator implementations in Hummingbird.
+    """
+
+    def __init__(self):
+        super().__init__()
+
+        # An operator can be either a model or a transformer. In the latter case, self.transformer must be set to True.
+        # In the former case, if a model is doing regression, then self.regression must be set to True. Otherwise the operator is doing some classificaiton task.
+        self.regression = False
```

Comment:
I think there are a lot of places that will be needed to be updated to use these, right?

Reply:
Nope, I was already doing this in the tree classes, I just moved all implementations to subclass this.